luthier/config/src/utils/defaultingUtils.lua

```diff
@@ -68,4 +71,17 @@ function M.logical_table_defaulting(logical_table_name, logical_table)
     logi_config.dateTimeZone = logi_config.dateTimeZone or M.DEFAULT_LOGICAL_DATE_TIME_ZONE
     return logi_config
 end
+
+function M.metric_defaulting(metric_name, metric)
+    local metric_config = misc.shallow_copy(metric)
+    metric_config.longName = metric_config.longName or metric_name
+    metric_config.description = metric_config.description or metric_config.longName   -- this line must follow the previous one
+    metric_config.category = metric_config.category or M.DEFAULT_METRIC_CATEGORY
+    metric_config.dataType = metric_config.dataType or M.DEFAULT_METRIC_DATA_TYPE
+    --- the following unification is no longer necessary in my (Gabriel's) opinion.
+    --- the discrepancy can be handled by two types of Factories: Aggregation and PostAggregation
+    --metric_config.dependencyMetricNames = metric_config.druidMetric and {metric_config.druidMetric}
+    --        or metric_config.dependencies       -- if druidMetric exists, use {druidMetric}, otherwise dependencies
+    return metric_config
```

Comment:
Actually, what really needs to happen is that the dataType should probably become optional.
The logical metric should define the default data type via the makers and the apimetricname should override that datatype if present.

My advice is to make the Factory treat this as an optional parameter and remove the defaulting altogether.

Reply:
addressed